src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/MetafileJsonValidatorTest.java

```diff
@@ -23,107 +24,107 @@ public void setUp() throws IOException, ProcessingException {
     public void should_parse_envelope_data_with_no_account_details_for_postal_payment() throws IOException {
-        Envelope envelope = getEnvelope("/metafiles/valid/postal-payment-method.json");
+        InputEnvelope envelope = getEnvelope("/metafiles/valid/postal-payment-method.json");
 
-        assertThat(envelope.getPayments()).hasSize(1);
-        assertThat(envelope.getPayments().get(0).getAmount()).isEqualTo(100.0);
-        assertThat(envelope.getPayments().get(0).getPaymentInstrumentNumber()).isEqualTo("1000000000");
-        assertThat(envelope.getPayments().get(0).getAccountNumber()).isNull();
-        assertThat(envelope.getPayments().get(0).getSortCode()).isNull();
+        assertThat(envelope.payments).hasSize(1);
+        assertThat(envelope.payments.get(0).amount).isEqualTo(new BigDecimal("100.00"));
+        assertThat(envelope.payments.get(0).paymentInstrumentNumber).isEqualTo("1000000000");
+        assertThat(envelope.payments.get(0).accountNumber).isNull();
+        assertThat(envelope.payments.get(0).sortCode).isNull();
     }
 
     @Test
     public void should_parse_envelope_data_with_no_payment_reference_number_for_cash_payment() throws IOException {
-        Envelope envelope = getEnvelope("/metafiles/valid/cash-payment-method.json");
+        InputEnvelope envelope = getEnvelope("/metafiles/valid/cash-payment-method.json");
 
-        assertThat(envelope.getPayments()).hasSize(1);
-        assertThat(envelope.getPayments().get(0).getAmount()).isEqualTo(100.0);
-        assertThat(envelope.getPayments().get(0).getPaymentInstrumentNumber()).isNull();
-        assertThat(envelope.getPayments().get(0).getAccountNumber()).isNull();
-        assertThat(envelope.getPayments().get(0).getSortCode()).isNull();
+        assertThat(envelope.payments).hasSize(1);
+        assertThat(envelope.payments.get(0).amount).isEqualTo(new BigDecimal("100.00"));
+        assertThat(envelope.payments.get(0).paymentInstrumentNumber).isNull();
+        assertThat(envelope.payments.get(0).accountNumber).isNull();
+        assertThat(envelope.payments.get(0).sortCode).isNull();
     }
 
     @Test
     public void should_parse_envelope_data_with_account_details_for_cheque_payment() throws IOException {
-        Envelope envelope = getEnvelope("/metafiles/valid/cheque-payment-method.json");
+        InputEnvelope envelope = getEnvelope("/metafiles/valid/cheque-payment-method.json");
 
-        assertThat(envelope.getPayments()).hasSize(1);
-        assertThat(envelope.getPayments().get(0).getAmount()).isEqualTo(100.0);
-        assertThat(envelope.getPayments().get(0).getPaymentInstrumentNumber()).isEqualTo("1000000000");
-        assertThat(envelope.getPayments().get(0).getAccountNumber()).isEqualTo("12345678");
-        assertThat(envelope.getPayments().get(0).getSortCode()).isEqualTo("112233");
+        assertThat(envelope.payments).hasSize(1);
+        assertThat(envelope.payments.get(0).amount).isEqualTo(new BigDecimal("100.00"));
```

Comment:
instead of using get(0) can we use something similar to
```
assertThat(fellowshipOfTheRing).extracting("name", "age", "race.name")
                               .contains(tuple("Boromir", 37, "Man"),
                                         tuple("Sam", 38, "Hobbit"),
                                         tuple("Legolas", 1000, "Elf"));
```

Reply:
Can be done like this. My change only replaced the use of getters with the use of public immutable fields. There are a few places where we can make this change and given the size of this PR, it would be best to have a separate one for that.

Reply:
👍